src/replication.c

```diff
@@ -2365,6 +2434,13 @@ handle_mgmt_conn_error(replica_t *r, int sfd, struct epoll_event *events, int ev
 		r->spec->target_replica = NULL;
 		r->spec->rebuild_in_progress = false;
 	}
+
+	TAILQ_FOREACH(kr, &r->spec->identified_replica, next) {
+		if (kr->zvol_guid == r->zvol_guid) {
+			kr->is_connected = false;
+			break;
+		}
+	}
 	MTX_UNLOCK(&r->spec->rq_mtx);
 
```

Comment:
we can add an ASSERT for debug build if kr is NULL

Reply:
ok.. may be, not required..
if too many replicas gets connected, there are chances that kr may be NULL